src/ipa_pc_seg/data_structures.rs

```diff
@@ -0,0 +1,366 @@
+use crate::*;
```

Comment:
Not a fan of this for several reasons:
1) It's difficult to review and track the segmentation-specific code;
2) Lot of code duplication;
3) Segmentation is an optimization of IPA and not a completely separated PC scheme;
4) I would like Marlin side to add another flag "SEGMENTATION" to the MarlinConfig trait that the user can enable or not, changing transparently the internal behaviour.

I think all of the above would be easier if we integrate the segmentation inside the already existing code, and we can do this in two ways:

1. You can create a SegmentedPolynomialCommitment trait mixing in also the PolynomialCommitment trait with the same (relevant) functions of the PolynomialCommitment trait. Whoever wants to adopt segmentation can implement this trait and override only the necessary functions according to the new logic;
2. We transparently use segmentation or not based on the size of the committer/verifier key: if the number of basis is smaller than the size of polynomial to commit we adopt the segmentation otherwise not. This requires to modify the existing data structures and functions to work with both logic (if not already possible).

Personally, I would like more solution 2, but we can discuss it separately.

Reply:
Variant 2 is done